config/routes.rb

```diff
@@ -1,10 +1,18 @@
 Rails.application.routes.draw do
-  resources :likes
-  resources :users
-  resources :comments
-  resources :posts
+  # resources :likes
+  # resources :users
+  # resources :comments
+  # resources :posts
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
-  root "users#index"
+  get '/users', to: 'users#index'
+  get '/users/:id', to: 'users#show', as: 'show_a_user'
+  
+  get '/users/:user_id/posts', to: 'posts#index', as: 'show_a_user_posts_comments'
+
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'show_a_user_a_post_comments'
+
+  root 'users#index'
```

Comment:
- [ ] You have done great adding the routes to the correct action, but as your project gets bigger, it will be a bit of a challenge handling the routes due to the way its currently written. I'll suggest you do this in a more conventional way by using `resources.` here's a perfect implementation example. :+1: :+1: 

```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

Reply:
Implemented